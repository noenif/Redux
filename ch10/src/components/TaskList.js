import React from 'react';
import Task from './Task';

const TaskList = props => {
  return (
    <div className="task-list">
      <div className="task-list-title">
        <strong>{props.status}</strong>
      </div>
      {props.taskIds.map(id => (
        <Task key={id} taskId={id} onStatusChange={props.onStatusChange} />
      ))}
    </div>
  );
};

export default TaskList;
