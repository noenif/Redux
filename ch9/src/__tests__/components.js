import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import TaskList from '../components/TaskList';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('the TaskList component', () => {
  it('should render a status', () => {
    const wrapper = shallow(<TaskList status="In Progress" tasks={[]} />);

    expect(wrapper.find('strong').text()).toEqual('In Progress');
  });

  it('should render a Task component for each task', () => {
    const tasks = [
      { id: 1, title: 'A', description: 'a', status: 'Unstarted', timer: 0 },
      { id: 2, title: 'B', description: 'b', status: 'Unstarted', timer: 0 },
      { id: 3, title: 'C', description: 'c', status: 'Unstarted', timer: 0 },
    ];
    const wrapper = shallow(<TaskList status="Unstarted" tasks={tasks} />);

    expect(wrapper.find('Task').length).toEqual(3);
  });

  it('should match the last snapshot without tasks', () => {
    const wrapper = shallow(<TaskList status="In Progress" tasks={[]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match the last snapshot with tasks', () => {
    const tasks = [
      { id: 1, title: 'A', description: 'a', status: 'Unstarted', timer: 0 },
      { id: 2, title: 'B', description: 'b', status: 'Unstarted', timer: 0 },
      { id: 3, title: 'C', description: 'c', status: 'Unstarted', timer: 0 },
    ];
    const wrapper = shallow(<TaskList status="In Progress" tasks={tasks} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
