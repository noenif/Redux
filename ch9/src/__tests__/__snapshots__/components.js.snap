// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the TaskList component should match the last snapshot with tasks 1`] = `
<div
  className="task-list"
>
  <div
    className="task-list-title"
  >
    <strong>
      In Progress
    </strong>
  </div>
  <Task
    key="1"
    task={
      Object {
        "description": "a",
        "id": 1,
        "status": "Unstarted",
        "timer": 0,
        "title": "A",
      }
    }
  />
  <Task
    key="2"
    task={
      Object {
        "description": "b",
        "id": 2,
        "status": "Unstarted",
        "timer": 0,
        "title": "B",
      }
    }
  />
  <Task
    key="3"
    task={
      Object {
        "description": "c",
        "id": 3,
        "status": "Unstarted",
        "timer": 0,
        "title": "C",
      }
    }
  />
</div>
`;

exports[`the TaskList component should match the last snapshot without tasks 1`] = `
<div
  className="task-list"
>
  <div
    className="task-list-title"
  >
    <strong>
      In Progress
    </strong>
  </div>
</div>
`;
